### PR TITLE
Desktop: improve navigation options on edit

### LINKED ISSUE #1445: Pictures shown when manually adding dives
### Describe the issue:

- [x] Bug
- [ ] Change request
- [ ] New feature request
- [x] Discussion request

### Issue long description:

When adding a dive, the dive photo tab shows the pictures of the previously selected dive(s). While this is technically correct (the dive selection is still valid), it seems weird from an UI perspective.

### Operating system:

```
$ uname -a
Linux laptop 4.15.0-23-generic #25-Ubuntu SMP Wed May 23 18:02:16 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
```
I doubt that this is OS-dependent.
### Subsurface version:

Current master.

### Steps to reproduce:
1) Go to photos-tab with dive having pictures.
2) CTRL-+.
### Current behavior:

Previous pictures are shown.
### Expected behavior:

No pictures shown, unless new pictures are added manually (?)

### PR DESCRIPTION
**Pull request by janmulder**
### Describe the pull request:

- [x] Bug fix
- [x] Functional change
- [ ] New feature
- [ ] Code cleanup
- [ ] Build system change
- [ ] Documentation change
- [ ] Language translation

### Pull request long description:
As described in the referenced issue, we where able to navigate to nonlogical static pages (like information, statistics. extra data) when adding a dive. These are output style pages that make no sense on edit or add. Further, disable access to some pages when entering edit mode.

Notice that the small change in file mainwindow.cpp is simply because this did not work at all, and became superfluous any way.

Signed-off-by: Jan Mulder <jlmulder@xs4all.nl>

### Related issues:
Fixes: #1445

### Release note:
Not sure... needed for this trivial thing?

### CODE REVIEW

**Comment by bstoeger** [comment]
Thanks, that has bothered me for the longest time and I never could motivate myself to dive into the tab thing.

> Not sure... needed for this trivial thing?

Perhaps not needed, but I think it would be a good idea, as it fixes a user-visible bug.

**Comment by dirkhh** [comment]
I would agree - this is a behavioral change. A good one, of course, but we should mention it.
Now let me look at the code...

File: desktop-widgets/tab-widgets/maintab.cpp [hunk 1]
```diff
@@ -339,6 +344,9 @@ void MainTab::enableEdition(EditMode newEditMode)
 	MainWindow::instance()->dive_list()->setEnabled(false);
 	MainWindow::instance()->setEnabledToolbar(false);
 	MainWindow::instance()->enterEditState();
+	ui.tabWidget->setTabEnabled(2, false);
+	ui.tabWidget->setTabEnabled(3, false);
+	ui.tabWidget->setTabEnabled(5, false);
```
Comment (dirkhh) [inline desktop-widgets/tab-widgets/maintab.cpp:349]: Simply pattern matching makes me wonder why we disable the fourth tab in one case and not the other - I guess I should actually read the code in context and try to figure it out myself...
It's the Media tab... hmm... I call 'bug'

Comment (janmulder) [reply desktop-widgets/tab-widgets/maintab.cpp:349]: As the media tab makes sense on edit :-)

Comment (dirkhh) [reply desktop-widgets/tab-widgets/maintab.cpp:349]: it does? How?

Comment (janmulder) [reply desktop-widgets/tab-widgets/maintab.cpp:349]: You can add media when editing a dive using the right mouse context menu ...

Comment (dirkhh) [reply desktop-widgets/tab-widgets/maintab.cpp:349]: I guess that shows you how often I use _THAT_ feature.

**Comment by dirkhh** [comment]
I hate to be a pain, but that CHANGELOG... hmm.
How about
`- Desktop: disable UI elements that make no sense during editing`

**Comment by bstoeger** [comment]
Tested this and it works nicely. The changes look very reasonable and it fixes a UI-problem. I think we can commit.

My only *very* minor gripe: When I'm on the statistics tab and edit the dive, it jumps to the media tab. I'd prefer if it goes to the notes tab.

**Comment by bstoeger** [comment]
There's another minor inconsistency, though I think it is not necessary to bother now, I just want it documented:

If in edit mode the user adds a picture and then says "Discard Changes", the picture is obviously still there.

**Comment by bstoeger** [comment]
Yes. My personal vision would be to incorporate all that into the undo-system. But that will be a steep uphill battle. :(

**Comment by janmulder** [comment]
@bstoeger  wrote: 
> My only very minor gripe: When I'm on the statistics tab and edit the dive, it jumps to the media tab. I'd prefer if it goes to the notes tab.

This intrigued me. How can you edit a dive from the statistics tab? Oh wait, using the menu log-edit dive. I have to admit that this was the very first time I ever used that menu entry.  In fact, I do not see _any_ reason for this menu option. @dirkhh: why do we need a menu option to edit a dive, when we can just start typing in a dive to be edited?

**Comment by bstoeger** [comment]
Heh. Just shows how different people are. :) Personally, I wouldn't have a problem with the entry being removed.

**Comment by dirkhh** [comment]
This is a left-over of a long past history, where you could't just edit things in place. Sure, we could remove that.

**Comment by janmulder** [comment]
Ok, then I opt for removing the edit dive from the menu, as fixing the navigation (probably also very simple) is like fixing something for a now dead piece of code. Will do that maybe today or ultimately tomorrow.

**Pull request merged**